src/vcpkg/paragraphs.cpp

```diff
@@ -379,106 +407,146 @@ namespace vcpkg::Paragraphs
-                                                                     bool is_manifest,
-                                                                     MessageSink& warning_sink)
+    ExpectedL<SourceControlFileAndLocation> try_load_control_file_text(StringView text,
+                                                                       StringView origin,
+                                                                       StringView spdx_location)
     {
         StatsTimer timer(g_load_ports_stats);
+        return parse_paragraphs(text, origin)
+            .then([&](std::vector<Paragraph>&& vector_pghs) -> ExpectedL<SourceControlFileAndLocation> {
+                auto maybe_parsed = SourceControlFile::parse_control_file(origin, std::move(vector_pghs));
+                if (auto parsed = maybe_parsed.get())
+                {
+                    return SourceControlFileAndLocation{
+                        std::move(*parsed), origin.to_string(), spdx_location.to_string()};
+                }
 
-        if (is_manifest)
-        {
-            return try_load_manifest_text(text, origin, warning_sink);
-        }
-
-        auto pghs = parse_paragraphs(StringView{text}, origin);
-        if (auto vector_pghs = pghs.get())
-        {
-            return SourceControlFile::parse_control_file(origin, std::move(*vector_pghs)).map_error(ToLocalizedString);
-        }
-
-        return std::move(pghs).error();
+                return ToLocalizedString(std::move(maybe_parsed).error());
+            });
     }
 
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_port(const ReadOnlyFilesystem& fs,
-                                                                StringView port_name,
-                                                                const Path& port_directory)
+    PortLoadResult try_load_port(const ReadOnlyFilesystem& fs, StringView port_name, const PortLocation& port_location)
     {
         StatsTimer timer(g_load_ports_stats);
 
-        const auto manifest_path = port_directory / "vcpkg.json";
-        const auto control_path = port_directory / "CONTROL";
+        const auto manifest_path = port_location.port_directory / "vcpkg.json";
+        const auto control_path = port_location.port_directory / "CONTROL";
         std::error_code ec;
         auto manifest_contents = fs.read_contents(manifest_path, ec);
         if (ec)
         {
-            const auto exists = ec != std::errc::no_such_file_or_directory;
-            if (exists)
+            auto manifest_exists = ec != std::errc::no_such_file_or_directory;
+            if (manifest_exists)
             {
-                return msg::format_error(msgFailedToParseManifest, msg::path = manifest_path)
-                    .append_raw("\n")
-                    .append(format_filesystem_call_error(ec, "read_contents", {manifest_path}));
+                return PortLoadResult{LocalizedString::from_raw(port_location.port_directory)
```

Comment:
@ras0219-msft suggests something like:

```
/long/path/to/my/zlib/abcd123515123123/vcpkg.json: error: permission denied
note: while loading port zlib
```

instead

Reply:
Consider splitting this yet further to avoid printing the path in the error message part